elastic/kibana
# [Infra UI] Propagate flyout state to node details page and some fixes

closes [#164300](https://github.com/elastic/kibana/issues/164300)

## Summary

This PR enables state propagation between asset details flyout and full page view.

https://github.com/elastic/kibana/assets/2767137/7e05a3c9-afa1-447c-98fd-91c40ee6cefb

There are other places in Kibana that redirect to node details outside the infra plugin such as APM and Observability/Overview. They use the `link-to/${assetType}-detail` path, so It's best, for now, to keep retro-compatibility with this route and propagate the state via query string.

I've also refactored how we were persisting state via route navigation, to use native the `state` attribute found in the `location` object from `react-router`

### How to tests
- Start a Kibana instance
- Navigate to `Infrastructure` > `Host`
  - Open the flyout, click in one of the tabs, and click on "Open as Page". The page should open in the same state as the flyout.
